### DISCUSSION TITLE
how to set cache control with spesific pathname/ url and static assets.

### DISCUSSION THREAD

**Discussion by isggwp** [vercel/next.js#21503] [Help] [answered]
Hello everyone, i've been setting for cache control in my custom server.js using setHeader like this:

```
// server.js
const { createServer } = require('http')
const { parse } = require('url')
const next = require('next')

const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev })
const handle = app.getRequestHandler()

app.prepare().then(() => {
  createServer((req, res) => {
 
      const parsedUrl = parse(req.url, true)
    

      res.setHeader(
        "Cache-Control",
        "max-age=0 s-maxage=60, stale-while-revalidate"
      );

      handle(req, res, parsedUrl);
  
  }).listen(3000, (err) => {
    if (err) throw err
    console.log('> Ready on http://localhost:3000')
  })
})

```

but, sometimes we need to setting  cache control to  **no-cache** or **stale-while-revalidate** for  dynamic content, and long cache to  **max-age=31536000**  for static assets inside static pages. 
can I deal without express for this? hope you give me warm advice. i just no have idea for this :)

**Comment by stigkj** [comment, marked as answer]
You can use this: https://nextjs.org/docs/api-reference/next.config.js/headers
